instrumentation/httpurlconnection/testing/src/androidTest/kotlin/io/opentelemetry/instrumentation/library/httpurlconnection/InstrumentationTest.kt

```diff
@@ -67,46 +70,39 @@ class InstrumentationTest {
             // Wait for all tasks to finish execution or timeout
             if (executor.awaitTermination(2, TimeUnit.SECONDS)) {
                 // if all tasks finish before timeout
-                Assert.assertEquals(4, inMemorySpanExporter.finishedSpanItems.size)
+                assertThat(inMemorySpanExporter.finishedSpanItems.size).isEqualTo(4)
             } else {
                 // if all tasks don't finish before timeout
-                Assert.fail(
+                fail(
                     "Test could not be completed as tasks did not complete within the 2s timeout period.",
                 )
             }
         } catch (e: InterruptedException) {
```

Comment:
Preference is to just let the exception escape the method and the test will still fail.